### ISSUE TITLE
eager mode fails with custom operator

### ISSUE THREAD

**Issue by sjperkins** [tensorflow/tensorflow#18455]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:

Yes, I have written a custom tensorflow operator

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:

Linux Ubuntu 16.04

- **TensorFlow installed from (source or binary)**:

binary wheel

- **TensorFlow version (use command below)**:

('v1.7.0-3-g024aecf414', '1.7.0')

- **Python version**: 

python 2.7.12

- **Bazel version (if compiling from source)**:

N/A

- **GCC/Compiler version (if compiling from source)**:

N/A

- **CUDA/cuDNN version**:

N/A

- **GPU model and memory**:

N/A

- **Exact command to reproduce**:

Custom operator is defined [here](https://github.com/ska-sa/montblanc/blob/f168870a9756f31c8639ff0c0219ce16af68e612/montblanc/impl/rime/tensorflow/rime_ops/phase_op_cpu.h#L44-L122) but I doubt you'll want to look through that.

```python
import tensorflow as tf
tf.enable_eager_execution()

from montblanc.impl.rime.tensorflow.tensorflow_ops import phase

lm = np.random.random((10,2)).astype(np.float32)
uvw = np.random.random((10,3,2)).astype(np.float32)
freq = np.linspace(.856e9, .856e8*2, 4).astype(np.float32)

# Convert into tfe Variable
lm, uvw, freq = (tfe.Variable(a) for a in (lm, uvw, freq))
value = phase(lm, uvw, freq, CT=tf.complex64)
```

This produces the following stack trace

```
Traceback (most recent call last):
  File "test_tf_eager_grad.py", line 65, in <module>
    value = phase(lm, uvw, freq, CT=tf.complex64)
  File "<string>", line 419, in phase
  File "/home/sperkins/venv/mb/local/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py", line 328, in apply_op
    op_type_name, name, **keywords)
  File "/home/sperkins/venv/mb/local/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/home/sperkins/venv/mb/local/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 3277, in create_op
    input_ops = set([t.op for t in inputs])
  File "/home/sperkins/venv/mb/local/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 842, in op
    raise AttributeError("op not supported for Eager Tensors.")
AttributeError: op not supported for Eager Tensors.
```

### Describe the problem

Eager mode does not work with a custom operator.

### Source code / logs

See above

**Comment by asimshankar** [comment]
Thanks for the report, this was indeed an oversight. @akshaym has a fix for this that should be merged soon.

**Comment by akshaym** [comment]
This should be fixed with #18504.

Thanks for the report.

**Comment by sjperkins** [comment]
Thanks @akshaym. Will this go into the 1.8 release?

**Comment by asimshankar** [comment]
@sjperkins : Alas, the fix was made after the 1.8 branch was cut, so this is not included there. It will be included in 1.9.
